src/EVM/Expr.hs

```diff
@@ -628,6 +633,32 @@ getAddr (GVar _) = error "cannot determine addr of a GVar"
 
 -- ** Whole Expression Simplification ** -----------------------------------------------------------
 
+-- Calls prop simplification for the appropriate types of Expr
+applyToProps :: ([Prop] -> [Prop]) -> Expr b -> Expr b
```

Comment:
do we actually need this? or can we just use `foldl'` and `fmap` combined with `mapProp`?

Reply:
Hmmm... `mapProp` doesn't apply this simplification:

```
  ITE (Lit 0) _ c -> applyToProps f c
  ITE (Lit _) b _ -> applyToProps f b
  ITE a b c-> ITE a (applyToProps f b) (applyToProps f c)
```

which does partial constant folding of `ITE`. The function `simplify` does apply this simplification, but that can't be called here, because it deals with a different type, right? Or is there something I'm missing? 

As for the rest, I don't even know how to do `fold'` and `fmap` such that it only applies to `ITE` and not the other things. That's what we'd need, right? Sorry, what you wrote is a bit mystical to me :D